src/components/Trade/TradeConfirm/TradeConfirm.tsx

```diff
@@ -98,21 +102,22 @@ export const TradeConfirm = ({ history }: RouterProps) => {
         validate: (txs: TradeTxs & { e: Error }) => !!txs.buyTxid || !!txs.e,
         interval: 10000, // 10 seconds
         maxAttempts: 300, // Lots of attempts because some trade are slow (thorchain to bitcoin)
+      }).then(txs => {
+        if (txs.e) throw txs.e
+        // we might not have the buyTxid right away
+        setBuyTxid(txs.buyTxid ?? '')
       })
-
-      if (txs.e) throw txs.e
-      if (!txs?.sellTxid) throw new Error('No sellTxid from getTradeTxs')
-      setTxid(txs.sellTxid)
+      setSellTxid(result.tradeId)
```

Comment:
I get it why we'd use `.then()` vs. `await`ing poll, as the former isn't blocking the execution of the following lines, but if we first `setSellTxId`, and `await poll()`, we should be able to use `async/await` syntax.

Reply:
The button spins because of `isSubmitting` on the formState.

The problem seems to be because its awaiting within the execution of the forms onSubmit function `isSubmitting` and the function therefor does not return until the polling completes so theresfor `isSubmitting` is stuck as set

Reply:
discovered a few more issues. putting back in draft